vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/InputFieldView.kt

```diff
@@ -982,67 +985,144 @@ abstract class InputFieldView @JvmOverloads constructor(
     }
 
     protected fun setOutputPattern(pattern: String?) {
-        if (fieldType == FieldType.CARD_EXPIRATION_DATE) {
-            (inputField as? DateInputField)?.setOutputPattern(pattern)
+        when (fieldType) {
+            FieldType.CARD_EXPIRATION_DATE, FieldType.DATE_RANGE -> {
+                (inputField as? DateInputField)?.setOutputPattern(pattern)
+            }
+            else -> {
+                // Do nothing
+            }
+        }
+    }
+
+    protected fun setDateFormat(pattern: String?) {
+        when (fieldType) {
+            FieldType.DATE_RANGE -> {
+                (inputField as? DateInputField)?.setInputFormat(pattern, VGSDateFormat.mmddyyyy())
```

Comment:
@DonaldRG Why it's hardcoded?

Reply:
Since DateInputField don't know anything about the field type (Expiration date or a Range date), and we need to set a default value for the date format, I'm sending it here as a parameter:
- VGSDateFormat.mmddyyyy is the default format for type DateRange
- VGSDateFormat.mmyy is the default format for ExpirationDate

Inside the method, if the pattern cannot be parsed or is invalid, the app will use the default format instead:
```
internal fun setInputFormat(pattern: String?, default: VGSDateFormat) {
    val parsedFormat = VGSDateFormat.parsePatternToDateFormat(pattern)
    inputFormat = when (parsedFormat) {
        null -> default
        else -> parsedFormat
    }
   ...
```